source/common/http/conn_manager_utility.cc

```diff
@@ -439,6 +439,10 @@ bool ConnectionManagerUtility::maybeNormalizePath(RequestHeaderMap& request_head
     return true; // It's as valid as it is going to get.
   }
   bool is_valid_path = true;
+  config.normalizePath(request_headers);
```

Comment:
Apologies if this has been already discussed. This looks confusing -- first `normalizePath()` and then `shouldNormalizePath()` check is made. Perhaps a comment re: cleaning up percent-encoded paths could be moved here?

Reply:
Thanks, will clean up the interface later.